controllers/budget_controller.py

```diff
@@ -1,76 +1,145 @@
+
+        sum_activity = sum_budgeted - sum_available
+
+        # Below code adds rows to PrettyTable like this:
+        # | -------------------- | -------------------- | -------------------- | -------------------- |
+        # | parent_category      |               xxx.xx |               yyy.yy |               zzz.zz |
+        # | -------------------- | -------------------- | -------------------- | -------------------- |
+        table_budget.add_row([30 * "-", 10 * "-", 10 * "-", 10 * "-"])
+        table_budget.add_row(
+            [ParentCategory.give_parent_categories(parent_instance), sum_budgeted, sum_activity, sum_available])
+        table_budget.add_row([30 * "-", 10 * "-", 10 * "-", 10 * "-"])
+
+        total_budgeted += sum_budgeted
+        total_activity += sum_activity
+        total_available += sum_available
+
+        sum_budgeted = 0  # this is reset to zero to allow summing categories in next parent-categories
+        sum_available = 0  # this is reset to zero to allow summing categories in next parent-categories
+
+        for category_instance in parent_instance.categories:
+            table_budget.add_row(Category.give_info(category_instance))
+
+        # Below code makes a visible break between parent categories (just for better readability of the table)
+        table_budget.add_row([" ", " ", " ", " "])
+
+    print(
+        f"\nHere is your budget \"{budget.name}\"")  # TODO: Here could be "Here is your budget called "{budget.name}"
+    print("-----------------------------")
+    print("MONTH: >>month<<")  # TODO: fill {month} with the right data (do f-string)
+    print("-----------------------------")
+    print(
+        f"TOTAL BUDGETED:   {round(total_budgeted, 2)}        TO BE BUDGETED:   >>to_be_budgeted<<")  # TODO: fill to_be_budgeted
+    print(f"TOTAL ACTIVITY:   {round(total_activity, 2)}")
+    print(f"TOTAL AVAILABLE:  {round(total_available, 2)}")
+
+    print(table_budget)
+
+
+def reports(budget_id):
+    print("\nREPORTS MENU:")
+    print("0. Print the table with all transactions in this budget   [FUNCTION NOT AVAILABLE YET]")
+    print("1. Display the bar chart")
+    print("2. Eksport my budget do csv file   [FUNCTION NOT AVAILABLE YET]")
+    print("3. Eksport my budget to json file  [FUNCTION NOT AVAILABLE YET]")
+    choice = input("## YOUR CHOICE: ")
+
+    if choice == "1":
+        # Definition of PrettyTable
+        bar_chart = PrettyTable()
+        bar_chart.field_names = [" id, CATEGORY", "ACTIVITY", "BAR CHART"]
+        bar_chart.align = "l"  # align in all columns to the left side
+        bar_chart.align["ACTIVITY"] = "r"  # align in column "ACTIVITY" to the right side
+        bar_chart.float_format = "1.2"  # the way floating point data is printed, for example "123.45"
+
+        # reading values from database and inserting into PrettyTable
+        for instance in session.query(ParentCategory).filter(ParentCategory.budget_id == budget_id):
```

Comment:
`for parent in budget.parent_categories` załatwiłoby tu sprawę :)

Reply:
poprawione